js/compiler.ts

```diff
@@ -541,6 +648,54 @@ window.compilerMain = function compilerMain(): void {
   };
 };
 
+const decoder = new TextDecoder();
+
+// Perform the op to retrieve the compiler configuration if there was any
+// provided on startup.
+function getCompilerConfig(
+  compilerType: string
+): { path: string; data: string } {
+  const builder = flatbuffers.createBuilder();
+  const compilerType_ = builder.createString(compilerType);
+  msg.CompilerConfig.startCompilerConfig(builder);
+  msg.CompilerConfig.addCompilerType(builder, compilerType_);
+  const inner = msg.CompilerConfig.endCompilerConfig(builder);
+  const baseRes = sendSync(builder, msg.Any.CompilerConfig, inner);
+  assert(baseRes != null);
+  assert(msg.Any.CompilerConfigRes === baseRes!.innerType());
+  const res = new msg.CompilerConfigRes();
+  assert(baseRes!.inner(res) != null);
+
+  // the privileged side does not normalize path separators in windows, so we
+  // will normalize them here
+  const path = res.path()!.replace(/\\/g, "/");
+  assert(path != null);
+  const dataArray = res.dataArray()!;
+  assert(dataArray != null);
+  const data = decoder.decode(dataArray);
+  return { path, data };
+}
+
 export default function denoMain(): void {
   os.start("TS");
+
+  const { path, data } = getCompilerConfig("typescript");
+  if (data.length) {
+    const ignoredOptions = compiler.configure(path, data);
+    if (ignoredOptions) {
+      if (os.noColor) {
+        console.warn(
+          `Unsupported compiler options in "${path}"\n` +
+            `  The following options were ignored:\n` +
+            `    ${ignoredOptions.join(", ")}`
+        );
+      } else {
+        console.warn(
+          `\x1b[33mUnsupported compiler options in "${path}"\x1b[39m\n` +
+            `  \x1b[36mThe following options were ignored:\x1b[39m\n` +
+            `    \x1b[1m${ignoredOptions.join("\x1b[22m, \x1b[1m")}\x1b[22m`
```

Comment:
I think it would be better to define a function like you have in std/colors which switches on bold, so this would look like something
```
console.warn(bold(`Unsupported compiler options in ${yellow(path)}`))
```
I'd say let's just start a minimal //js/colors.ts module, with a TODO to remove it and use std/colors.ts once we're able to import it.